GameFiles/_integration.gsc

```diff
@@ -490,14 +505,59 @@ NotifyClientEvent( eventInfo )
     if ( level.iw4adminIntegrationDebug == 1 )
     {
         IPrintLn( "NotifyClientEvent->" + event.data );
+        if(int(eventInfo[3]) != -1 && !isDefined(origin))
+        {
+            IPrintLn("origin is null but the slot id is " + int(eventInfo[3]));
+        }
+        if(int(eventInfo[4]) != -1 && !isDefined(target))
+        {
+            IPrintLn("target is null but the slot id is " + int(eventInfo[4]));
+        }
     }
 
-    client = event.origin;
+    if(isDefined(target))
+    {
+        client = event.target;
+    }
+    else if(isDefined(origin))
+    {
+        client = event.origin;
+    }
+    else
+    {
+        if ( level.iw4adminIntegrationDebug == 1 )
+        {
+            IPrintLn( "Neither origin or target are set but we are a Client Event, aborting" );
+        }
+        return;
+    }
     client.event = event;
 
     level notify( level.eventTypes.localClientEvent, client );
 }
 
+getPlayerFromClientNum( clientNum )
+{
+	if ( clientNum < 0 )
+		return undefined;
+	
+	for ( i = 0; i < level.players.size; i++ )
+	{
+		if ( level.players[i] getEntityNumber() == clientNum )
+			return level.players[i];
+	}
+	return undefined;
+}
+
+addClientCommand(cmd, runAsTarget, callback, overwrite)
```

Comment:
nit: for argument names: `commandName, shouldRunAsTarget, callback, shouldOverwrite`